### PR TITLE
X509Store_MachineStoreLoadSkipsInvalidFiles: skip test when running as root

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/dotnet/corefx/issues/30162

CC @ViktorHofer

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs [hunk 1]
```diff
@@ -539,6 +539,11 @@ public static void UnixCannotModifyDisallowedStore(bool useEnum, OpenFlags openF
         [PlatformSpecific(TestPlatforms.Linux)] // Windows/OSX doesn't use SSL_CERT_{DIR,FILE}.
         private void X509Store_MachineStoreLoadSkipsInvalidFiles()
         {
+            if (geteuid() == 0)
+            {
+                throw new SkipTestException("Test can't be run as root"); // root can read '2.pem'
```
Comment (stephentoub) [inline src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs]: You need to change the [Fact] to be [ConditionalFact] for SkipTestException to have the desired effect.

Comment (stephentoub) [reply src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs]: Though I wonder if we should instead have:
```C#
public static bool NotRunningAsRoot => geteuid() != 0;
```
and then do:
```C#
[ConditionalFact(nameof(NotRunningAsRoot))]
```
to be more declarative about it.

Comment (ViktorHofer) [reply src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs]: Yes please only use SkipTestException if it's absolutely necessary. A conditional fact should probably suffice here.

Comment (tmds) [reply src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs]: Fixed!

**Pull request merged**